Added combine option to moviepy backend.

Sv3n opened the pull request:
Also, allow re-using previously generated files with flag use_existing.
backend.Run now returns a potentially changed filename (required for
audio inputs which transform to mp4 or some other video format).

Sv3n committed:
Added combine option to moviepy backend.

Also, allow re-using previously generated files with flag use_existing.
backend.Run now returns a potentially changed filename (required for
audio inputs which transform to mp4 or some other video format).

CNugteren commented on pypopquiz/io.py:
```diff
@@ -25,6 +25,7 @@ def verify_input(input_data: Dict) -> None:
             "round": {"type": "number"},
             "theme": {"type": "string"},
             "spacers": {"type": "boolean"},
+            "use_existing": {"type": "boolean"},
```
Can we give this a slightly more descriptive name? Something like `use_cached_video_files`? But still then it might also apply to the downloaded YouTube videos, but I think `use_existing` is a bit too general.

Sv3n committed:
Renamed use_existing flag, made it more robust.

Video files are now generated in a temporary folder, and only moved to
the output folder when done. If use_cached_video_files is set, and the
target file already exists in the output folder, then it is not
regenerated.

Output file type is fored to mp4 for now, to allow the "exists()" check
to work. This could be made a parameter later if required.

Sv3n commented on pypopquiz/video.py:
```diff
@@ -80,15 +80,21 @@ def create_video(kind: str, round_id: int, question: Dict, question_id: int, out
     stream = backend_cls(video_file, width=width, height=height)
-    stream_f = filter_stream(stream, kind, round_id, question, question_id, add_spacer=add_spacer)
-    file_name_out = stream_f.run(file_name, dry_run=use_existing)
+    stream = filter_stream(stream, kind, round_id, question, question_id, add_spacer=add_spacer)
```
I changed this to make the linter happy (too many local variables :/)

CNugteren replied:
Yeah the linter is getting annoying ;-) You can also increase the allowed number of local variables in `.pylintrc`.

Sv3n committed:
Added str conversion to Path.

CNugteren commented:
Thanks for adding the temporary folder thingy. I'll merge it in now and also test myself later.
